### PR TITLE
fix: refacto permissions screens

### PR DESCRIPTION
**Pull request by clegirar**
Signed-off-by: clegirar <clemntgirard@gmail.com>

### CODE REVIEW

File: js/packages/components/chat/footer/recorder/RecorderButton.tsx [hunk 1]
```diff
@@ -145,21 +128,30 @@ export const RecorderButton: React.FC<RecorderButtonProps> = ({
...
-		setCurrentTime(Date.now())
-		setClearRecordingInterval(setInterval(() => setCurrentTime(Date.now()), 100))
-		meteredValuesRef.current = []
+		await checkPermission({
+			permissionType: PermissionType.audio,
+			navigate,
+			accept: () => {
+				// We check the status before ask it, to don't record after asking permission
+				// (In other apps, the recorder don't start after accept permission)
+				if (status === RESULTS.GRANTED) {
```
Comment (n0izn0iz) [inline js/packages/components/chat/footer/recorder/RecorderButton.tsx:139]: I think `status` is out of date here

File: js/packages/components/chat/footer/CameraButton.tsx [hunk 2]
```diff
@@ -1,104 +1,29 @@
...
-
 	return (
 		<ChatInputButton
 			iconName='camera'
 			iconPack='custom'
 			vOffset={1.5}
 			iconRatio={0.44}
-			onPress={handlePressCamera}
+			onPress={async () => {
+				await handlePressCamera({ sending, setSending, messengerClient, onClose, navigate })
```
Comment (n0izn0iz) [inline js/packages/components/chat/footer/CameraButton.tsx:25]: no need to use async/await here I think

File: js/packages/utils/permissions/handle-press-permissions/prepareMediaAndSend.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+export const prepareMediaAndSend: (props: PrepareMediaAndSendProps) => void = async ({
+	sending,
+	setSending,
+	messengerClient,
+	onClose,
+	res,
+}) => {
+	try {
+		if (sending) {
```
Comment (n0izn0iz) [inline js/packages/utils/permissions/handle-press-permissions/prepareMediaAndSend.ts:16]: this might be out of date

**Review by n0izn0iz — APPROVED** [review]
LGTM except comments

File: js/packages/components/chat/footer/CameraButton.tsx (hunk 2 above)
Comment (clegirar) [reply js/packages/components/chat/footer/CameraButton.tsx:25]: Yes we need, in `handlePressCamera` function we have `checkPermission` await function that is necessary to await 👍

File: js/packages/components/chat/footer/recorder/RecorderButton.tsx (hunk 1 above)
Comment (clegirar) [reply js/packages/components/chat/footer/recorder/RecorderButton.tsx:139]: What do you mean by out of date ?

File: js/packages/utils/permissions/handle-press-permissions/prepareMediaAndSend.ts (hunk 3 above)
Comment (clegirar) [reply js/packages/utils/permissions/handle-press-permissions/prepareMediaAndSend.ts:16]: What do you mean by out of date ?

File: js/packages/components/chat/footer/CameraButton.tsx (hunk 2 above)
Comment (n0izn0iz) [reply js/packages/components/chat/footer/CameraButton.tsx:25]: using `() => handlePressCamera({ sending, setSending, messengerClient, onClose, navigate })` will return a promie and is the same as wrapping it in async/await

File: js/packages/components/chat/footer/recorder/RecorderButton.tsx (hunk 1 above)
Comment (n0izn0iz) [reply js/packages/components/chat/footer/recorder/RecorderButton.tsx:139]: I mean that by the time the `accept` callback is called, the status value might be different

File: js/packages/utils/permissions/handle-press-permissions/prepareMediaAndSend.ts (hunk 3 above)
Comment (n0izn0iz) [reply js/packages/utils/permissions/handle-press-permissions/prepareMediaAndSend.ts:16]: for example here: https://github.com/berty/berty/pull/4316/files/d4a1108ee45312079635f76887c02a6670915ae9#diff-a4fa0c07ae99f80ca0174f3eed339c9d3abb1588e397e5d012c7a517d5666aa1R32

by the time this part is run, the value of sending might be different

File: js/packages/components/chat/footer/recorder/RecorderButton.tsx (hunk 1 above)
Comment (clegirar) [reply js/packages/components/chat/footer/recorder/RecorderButton.tsx:139]: Mhhh, but this value is usefull just at the init

Comment (clegirar) [reply js/packages/components/chat/footer/recorder/RecorderButton.tsx:139]: But i want the `status` value before call `accept` callback

Comment (clegirar) [reply js/packages/components/chat/footer/recorder/RecorderButton.tsx:139]: I can change his name to be more clear like `initialStatus` something like that

**Pull request merged**